modules/client/src/main/java/org/apache/ignite/internal/client/table/ClientTable.java

```diff
@@ -839,25 +868,202 @@ <E> CompletableFuture<List<E>> split(
                         idx++;
                     }
 
-                    List<CompletableFuture<List<E>>> res = new ArrayList<>(aff.size());
-                    List<Batch<E>> batches = new ArrayList<>();
+                    CompletableFuture<List<E>> resFut = new CompletableFuture<>();
+                    mapAndRetry(fun, keys, txns, mapped, new long[1], resFut, log);
+                    return resFut;
+                });
+    }
+
+    private static <R, E> void mapAndRetry(
```

Comment:
I think we can move those new methods to a separate class dedicated to batch tx mapping. `ClientTable` is already too long.

Reply:
🆗